lightningd/options.c

```diff
@@ -210,6 +210,18 @@ static char *opt_set_fee_rates(const char *arg, struct chain_topology *topo)
 	return NULL;
 }
 
+static char *opt_set_pidfile(const char *arg, struct lightningd *ld)
+{
+        ld->pidfile = tal_strdup(ld, arg);
+        return NULL;
+}
+
+static void opt_show_pidfile(char buf[OPT_SHOW_LEN],
+                             const struct lightningd *ld)
+{
+        snprintf(buf, OPT_SHOW_LEN, "%s", ld->pidfile);
```

Comment:
This is subtle.  We actually want the default pid filename to change between mainnet etc (we also want the db and hsm filenames to change, but that's a separate issue we should address!), so the best way is to start it NULL, and then generate it as "lightningd-networkname.pid" before we use it.

This means opt_show_pidfile() should also use the same logic, generating the name if it's NULL.